pkg/auth/manager.go

```diff
@@ -20,11 +23,23 @@ type authManager struct {
 	authHandlers    map[policy.AuthType]authHandler
 }
 
+// authHandler is responsible to handle authentication for a specific auth type
 type authHandler interface {
-	authenticate() error
+	authenticate(*authRequest) (*authResponse, error)
```

Comment:
i'm not so sure about this as well. as mentioned, this would result in some retries as long as auth isn't finished anyway. 

i decided to keep the context of this PR around the "data" in the API and how to gather them (from `IPCache`).

but definitely needs some follow up.

Reply:
It's my first time looking at this code and I understand that this has already been partially implemented before, but I'd like to stress that `NotifyPerfEvent` _must not block_ the monitor producer. 

Check the other two implementations of `NotifyPerfEvent` and you'll see that both do some basic preprocessing and then shove the message in a queue to be processed asynchronously.

Otherwise we'll drop events that Hubble etc would like to observe while we're waiting for `NotifyPerfEvent` to process the event.

Reply:
@gandro thanks for pointing that out. i wasn't aware that this would also block also monitor consumers or affect dropping events :+1: 

I added a queue within the auth monitor consumer. keeping the asynchronous part within the monitor itself for the moment - e.g. without exposing the channel to the manager. might be a next step if necessary.